app/runtime/runtime/src/main/java/io/syndesis/integration/runtime/LoggingInterceptStrategy.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.integration.runtime;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import io.syndesis.core.KeyGenerator;
+import org.apache.camel.CamelContext;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.model.LogDefinition;
+import org.apache.camel.model.ProcessorDefinition;
+import org.apache.camel.spi.InterceptStrategy;
+import org.apache.camel.spi.LogListener;
+import org.apache.camel.spi.UuidGenerator;
+import org.apache.camel.util.CamelLogger;
+
+import java.io.StringWriter;
+
+/**
+ * Created by chirino on 1/10/18.
+ */
+public class LoggingInterceptStrategy implements InterceptStrategy {
+
+
+    public UuidGenerator getUuidGenerator() {
+        return new UuidGenerator() {
+            @Override
+            public String generateUuid() {
+                // Generates always incrementing lexically sortable unique keys
+                return KeyGenerator.createKey();
```

Comment:
why is 'lexicable sortable keys' required ?  Why not a plain java.util.UUID ?

Reply:
this key is used to insert and order transactions in the db without needing a 2nd index based on time.